cmake/modules/FindFFMPEG.cmake

```diff
@@ -358,6 +212,29 @@ else()
      OR WIN32)
     set(FFMPEG_VERSION ${REQUIRED_FFMPEG_VERSION})
 
+    # macro for find_library usage
+    # arg1: lowercase libname (eg libavcodec, libpostproc, etc)
+    macro(ffmpeg_find_lib libname)
+      string(TOUPPER ${libname} libname_UPPER)
+      string(REPLACE "lib" "" name ${libname})
+
+      find_library(FFMPEG_${libname_UPPER}
+                   NAMES ${name} ${libname}
+                   PATH_SUFFIXES ffmpeg/${libname}
+                   HINTS ${DEPENDS_PATH}/lib ${PC_FFMPEG_${libname}_LIBDIR}
```

Comment:
is ${DEPENDS_PATH}/lib really needed here? And if yes, shouldn't it be after the PC_FFMPEG_${libname}_LIBDIR?
Otherwise we might find it in depends even when building with external ffmpeg

Reply:
Yeah it is for passing possible search paths for everything non linux. DEPENDS_PATH is set for all other platforms, and pkg-config isnt used for windows at all.

On linux, DEPENDS_PATH (if not set by user) is set to the cmake build directory. If someone is building internal, then expecting to use the system version, they are going about things the wrong way, and i think its too much of an edge case to factor in that. Their solution will be wipe out the build directory, dont rebuild internal ffmpeg. It will then fall through to the pkg-config search paths